pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeSegmentDataManager.java

```diff
@@ -1758,6 +1769,12 @@ private void updateCurrentDocumentCountMetrics() {
       _segmentLogger.info(
           "Consumed {} events from (rate:{}/s), currentOffset={}, numRowsConsumedSoFar={}, numRowsIndexedSoFar={}",
           rowsConsumed, consumedRate, _currentOffset, _numRowsConsumed, _numRowsIndexed);
+      if (_filteredMessageOffsets.size() > 0) {
+        if (_trackFilteredMessageOffsets) {
```

Comment:
This check is redundant

Reply:
I left it on purpose if anyone changes the logic to pushing to _filteredMessageOffsets in the future we can still ensure that the config is enabled.